app/src/main/kotlin/com/softteco/template/utils/CryptoManager.kt

```diff
@@ -0,0 +1,84 @@
+package com.softteco.template.utils
+
+import android.security.keystore.KeyGenParameterSpec
+import android.security.keystore.KeyProperties
+import java.io.InputStream
+import java.io.OutputStream
+import java.security.KeyStore
+import javax.crypto.Cipher
+import javax.crypto.KeyGenerator
+import javax.crypto.SecretKey
+import javax.crypto.spec.IvParameterSpec
+
+class CryptoManager {
+
+    private val keyStore = KeyStore.getInstance(KEY_STORE_TYPE).apply {
+        load(null)
+    }
+
+    private val encryptCipher get() = Cipher.getInstance(TRANSFORMATION).apply {
+        init(Cipher.ENCRYPT_MODE, getKey())
+    }
+
+    private fun getDecryptCipherForIv(iv: ByteArray): Cipher {
+        return Cipher.getInstance(TRANSFORMATION).apply {
+            init(Cipher.DECRYPT_MODE, getKey(), IvParameterSpec(iv))
+        }
+    }
+
+    private fun getKey(): SecretKey {
+        val existingKey = keyStore.getEntry(KEY_STORE_ALIAS, null) as? KeyStore.SecretKeyEntry
+        return existingKey?.secretKey ?: createKey()
+    }
+
+    private fun createKey(): SecretKey {
+        return KeyGenerator.getInstance(ALGORITHM).apply {
+            init(
+                KeyGenParameterSpec.Builder(
+                    KEY_STORE_ALIAS,
+                    KeyProperties.PURPOSE_ENCRYPT or KeyProperties.PURPOSE_DECRYPT
+                )
+                    .setBlockModes(BLOCK_MODE)
+                    .setEncryptionPaddings(PADDING)
+                    .setUserAuthenticationRequired(false)
+                    .setRandomizedEncryptionRequired(true)
+                    .build()
+            )
+        }.generateKey()
+    }
+
+    fun encrypt(bytes: ByteArray, outputStream: OutputStream): ByteArray {
+        val encryptedBytes = encryptCipher.doFinal(bytes)
+        outputStream.use {
+            it.write(encryptCipher.iv.size)
+            it.write(encryptCipher.iv)
+            it.write(encryptedBytes.size)
+            it.write(encryptedBytes)
+        }
+        return encryptedBytes
```

Comment:
do we need handle exceptions?  (InvalidKeyException or BadPaddingException)

Reply:
@EvgeniyFilippov any updates here?

Reply:
I think it's not necessary to be able to manually intercept and handle errors when writing/reading to try/catch blocks at the viewmodel level.